src/types/pcm.d.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Products
+ * Description: Products are the core resource to any Commerce Cloud project. They can be associated by category, collection, brands, and more.
+ */
+import {
+  Identifiable,
+  CrudQueryableResource
+} from './core'
+
+/**
+ * Core PCM Product Base Interface
+ * For custom flows, extend this interface
+ */
+export interface PcmProductBase {
+  type: string
+  attributes: {
+    name: string
+    description?: string
+    slug: string
+    sku: string
+    status?: string
+    commodity_type?: string
+    upc_ean?: number
+    mpn?: string
+  }
+}
+
+export interface PcmProduct extends Identifiable, PcmProductBase {
+  meta: {
+    // TODO
+  }
+}extends/base.js
+
+
+export interface PcmProductFilter {
+  // TODO
+}
```

Comment:
Dont know what any of these filters/relationships look like, but I imagine we will continuously update this as the API comes together.